spec/views/_dataset_spec.rb

```diff
@@ -33,6 +33,27 @@
     expect(page.css('tr')[0].css('td')[2].inner_text).to match(/Yes/)
   end
 
+  def expect_columns(page)
+    expect(page.css('tr')[0].css('td').count).to eq(8)
+  end
+
+  it 'displays warning icon for URL inaccessible dataset' do
+    deprecated_date = DateTime.now
+    @dataset.update_column(:url, "http://www.deadurl.com/example.csv")
+    @dataset.update_column(:url_deprecated_at, deprecated_date)
+    render :partial => 'datasets/dataset.html.erb', :locals => {:dataset => @dataset}
+    page = Nokogiri::HTML(rendered)
+    expect(page.css('tr:first-child > td:nth-child(2)')).to have_css('i.fa.fa-exclamation-triangle');
+  end
+
+  it 'displays deprecation date when in the dashboard' do
+    @dashboard = true
+    render :partial => 'datasets/dataset.html.erb', :locals => {:dataset => @dataset}
+    page = Nokogiri::HTML(rendered)
+    expect_columns(page)
+    expect(DateTime.parse(page.css('tr')[0].css('td')[5].inner_text).instance_of?(DateTime))
```

Comment:
Is it worth checking for exact date text here? You can keep a specific time using `Timecop.freeze`. Not compulsory, you know you've got a date, just wondering if you want to be sure about the format. Up to you.